pkg/controllermanager/controller/seed/seed_lifecycle_control.go

```diff
@@ -0,0 +1,116 @@
+	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/logger"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+)
+
+func (c *Controller) seedAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+	c.seedQueue.Add(key)
+}
+
+func (c *Controller) reconcileSeedKey(key string) error {
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	seed, err := c.seedLister.Get(name)
+	if apierrors.IsNotFound(err) {
+		logger.Logger.Infof("[SEED LIFECYCLE] Stopping lifecycle operations for Seed %s since it has been deleted", key)
+		c.seedQueue.Done(key)
+		return nil
+	}
+	if err != nil {
+		logger.Logger.Infof("[SEED LIFECYCLE] %s - unable to retrieve object from store: %v", key, err)
+		return err
+	}
+
+	if err := c.control.Reconcile(seed, key); err != nil {
+		return err
+	}
+
+	c.seedQueue.AddAfter(key, 10*time.Second)
+	return nil
+}
+
+// ControlInterface implements the control logic for managing the lifecycle for Seeds. It is implemented as an interface to allow
+// for extensions that provide different semantics. Currently, there is only one implementation.
+type ControlInterface interface {
+	Reconcile(seed *gardencorev1alpha1.Seed, key string) error
+}
+
+// NewDefaultControl returns a new instance of the default implementation ControlInterface that
+// implements the documented semantics for checking the lifecycle for Seeds. You should use an instance returned from NewDefaultControl()
+// for any scenario other than testing.
+func NewDefaultControl(k8sGardenClient kubernetes.Interface, k8sGardenCoreInformers gardencoreinformers.Interface, config *config.ControllerManagerConfiguration) ControlInterface {
+	return &defaultControl{k8sGardenClient, k8sGardenCoreInformers, config}
+}
+
+type defaultControl struct {
+	k8sGardenClient        kubernetes.Interface
+	k8sGardenCoreInformers gardencoreinformers.Interface
+	config                 *config.ControllerManagerConfiguration
+}
+
+func (c *defaultControl) Reconcile(seedObj *gardencorev1alpha1.Seed, key string) error {
+	var (
+		seed       = seedObj.DeepCopy()
+		seedLogger = logger.NewFieldLogger(logger.Logger, "seed", seed.Name)
+	)
+
+	// New seeds don't have conditions, i.e., gardenlet never reported anything yet. Let's wait until it sends a heart beat.
+	if len(seed.Status.Conditions) == 0 {
+		return nil
+	}
+
+	for _, condition := range seed.Status.Conditions {
+		if condition.Type == gardencorev1alpha1.SeedGardenletReady && (condition.Status == gardencorev1alpha1.ConditionUnknown || !condition.LastUpdateTime.UTC().Before(time.Now().UTC().Add(-40*time.Second))) {
```

Comment:
I thought about it, but I couldn’t image a proper use-case for this. Do you have one?

Reply:
Yes. We identify different network latency over different cloud provider.   Specifically, tomorrow if we are going for managing bare-metal/private cloud solutions, then we might have additional middleware like proxy, firewall and vpn  between SAP Garden-apiserver and gardenlet running inside host cloud provider. This will add up the latency. So, let's make it configurable. 

If you don't find this sensible enough now, i am fine with postponing change later.